main.tf

```diff
@@ -0,0 +1,183 @@
+  resource_group_name              = "${var.resource_group_name}"
+  network_interface_ids            = ["${element(azurerm_network_interface.instance_nic.*.id, count.index)}"]
+  availability_set_id              = "${element(azurerm_availability_set.instance_av_set.*.id, 0)}"
+  vm_size                          = "${var.vm_size}"
+  count                            = "${var.num}"
+  delete_os_disk_on_termination    = true
+  delete_data_disks_on_termination = true
+
+  storage_image_reference {
+    publisher = "${contains(keys(var.image), "id") ? "" : module.dcos-tested-oses.azure_publisher}"
+    offer     = "${contains(keys(var.image), "id") ? "" : module.dcos-tested-oses.azure_offer}"
+    sku       = "${contains(keys(var.image), "id") ? "" : module.dcos-tested-oses.azure_sku}"
+    version   = "${contains(keys(var.image), "id") ? "" : module.dcos-tested-oses.azure_version}"
+    id        = "${lookup(var.image, "id", "")}"
+  }
+
+  storage_os_disk {
+    name              = "os-disk-${format(var.hostname_format, count.index + 1, local.cluster_name)}"
+    caching           = "ReadWrite"
+    create_option     = "FromImage"
+    managed_disk_type = "${var.disk_type}"
+    os_type           = "windows"
+  }
+
+  storage_data_disk {
+    name            = "${azurerm_managed_disk.instance_managed_disk.*.name[count.index]}"
+    managed_disk_id = "${azurerm_managed_disk.instance_managed_disk.*.id[count.index]}"
+    create_option   = "Attach"
+    caching         = "None"
+    lun             = 0
+    disk_size_gb    = "${azurerm_managed_disk.instance_managed_disk.*.disk_size_gb[count.index]}"
+  }
+
+  os_profile {
+    computer_name  = "${format(var.hostname_format, count.index + 1, local.cluster_name)}"
+    admin_username = "${local.admin_username}"
+    admin_password = "${element(random_password.password.*.result, count.index)}"
+    custom_data    = "${var.custom_data}"
+  }
+
+  os_profile_windows_config {
+    provision_vm_agent        = true
+    enable_automatic_upgrades = false
+  }
+
+  tags = "${merge(var.tags, map("Name", format(var.hostname_format, (count.index + 1), var.location, local.cluster_name),
+                                "Cluster", local.cluster_name))}"
+}
+
+resource "azurerm_virtual_machine_extension" "winrm_setup" {
+  name                 = "${format(var.hostname_format, count.index + 1, local.cluster_name)}"
+  location             = "${var.location}"
+  resource_group_name  = "${var.resource_group_name}"
+  virtual_machine_name = "${format(var.hostname_format, count.index + 1, local.cluster_name)}"
+  publisher            = "Microsoft.Compute"
+  type                 = "CustomScriptExtension"
+  type_handler_version = "1.8"
+  count                = "${var.num}"
+  depends_on           = ["azurerm_virtual_machine.windows_instance"]
+
+  settings = <<SETTINGS
+    {
+        "fileUris": ["https://raw.githubusercontent.com/dcos-terraform/terraform-azurerm-windows-instance/master/winrm_setup.ps1"],
```

Comment:
also as said in #1 
please use the file operation here for commandToExecute as we could not bet on github being available and want exactly the version of the script we package in each specific terraform module release.

Reply:
Tried `"fileUris": ["file(${path.module/winrm_script.ps1})"]` and it doesn't work . 
Reason: as per [Microsoft documentation](https://docs.microsoft.com/en-us/azure/virtual-machines/extensions/custom-script-windows) - "Scripts can be downloaded from Azure storage or GitHub, or provided to the Azure portal at extension run time." Meaning, solution here is either to place content of winrm script using `cmd` syntax and execute line by line, or use as is - via github link, as shown in the example

Reply:
I see but this is a no go, as it is not bound to a released terraform module then.
option would to add it via cmd and read by `template` or `file` to be interpreted via terraform.

Reply:
removed `fileUris` permanently
added `commandToExecute` with all command lines at `winrm_setup.ps1`
removed `winrm_setup.ps1`